lib/core/src/sync/model/mod.rs

```diff
@@ -0,0 +1,312 @@
+    pub(crate) preimage: String,
+    pub(crate) create_response_json: String,
+    pub(crate) claim_fees_sat: u64,
+    pub(crate) claim_private_key: String,
+    pub(crate) payer_amount_sat: u64,
+    pub(crate) receiver_amount_sat: u64,
+    pub(crate) created_at: u32,
+    pub(crate) payment_hash: Option<String>,
+    pub(crate) description: Option<String>,
+}
+
+impl From<ReceiveSwap> for ReceiveSyncData {
+    fn from(value: ReceiveSwap) -> Self {
+        Self {
+            swap_id: value.id,
+            payment_hash: value.payment_hash,
+            invoice: value.invoice,
+            preimage: value.preimage,
+            create_response_json: value.create_response_json,
+            claim_fees_sat: value.claim_fees_sat,
+            claim_private_key: value.claim_private_key,
+            payer_amount_sat: value.payer_amount_sat,
+            receiver_amount_sat: value.receiver_amount_sat,
+            created_at: value.created_at,
+            description: value.description,
+        }
+    }
+}
+
+impl From<ReceiveSyncData> for ReceiveSwap {
+    fn from(val: ReceiveSyncData) -> Self {
+        ReceiveSwap {
+            id: val.swap_id,
+            payment_hash: val.payment_hash,
+            preimage: val.preimage,
+            create_response_json: val.create_response_json,
+            claim_private_key: val.claim_private_key,
+            invoice: val.invoice,
+            description: val.description,
+            payer_amount_sat: val.payer_amount_sat,
+            receiver_amount_sat: val.receiver_amount_sat,
+            claim_fees_sat: val.claim_fees_sat,
+            created_at: val.created_at,
+            claim_tx_id: None,
+            state: PaymentState::Created,
+            is_local: false,
+        }
+    }
+}
+
+#[derive(Serialize, Deserialize)]
+#[serde(tag = "data_type", content = "data")]
+pub(crate) enum SyncData {
+    Chain(ChainSyncData),
+    Send(SendSyncData),
+    Receive(ReceiveSyncData),
+}
+
+impl SyncData {
+    pub(crate) fn to_bytes(&self) -> serde_json::Result<Vec<u8>> {
+        serde_json::to_vec(self)
+    }
+}
+
+pub(crate) struct DecryptedRecord {
+    pub(crate) id: String,
+    pub(crate) revision: i64,
+    pub(crate) schema_version: f32,
+    pub(crate) data: SyncData,
+}
+
+impl DecryptedRecord {
+    pub(crate) fn try_from_record(
+        signer: Arc<Box<dyn Signer>>,
+        record: &Record,
+    ) -> anyhow::Result<Self> {
+        let dec_data = signer.ecies_decrypt(record.data.as_slice())?;
+        let data = serde_json::from_slice(&dec_data)?;
+        Ok(Self {
+            id: record.id.clone(),
+            revision: record.revision,
+            schema_version: record.schema_version,
+            data,
+        })
+    }
+}
+
+impl Record {
+    pub(crate) fn new(
+        data: SyncData,
+        revision: Option<i64>,
+        signer: Arc<Box<dyn Signer>>,
+    ) -> Result<Self, anyhow::Error> {
+        let id = uuid::Uuid::new_v4().to_string();
```

Comment:
It seems that we always create a new id and not saving it in the db (associate it with the swap).
the whole point of this client generated id is so we can associate later updated data to the right record in the db.